src/persist-client/src/batch.rs

```diff
@@ -497,6 +498,19 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
         }
     }
 
+    pub(crate) async fn flush(&mut self) -> Vec<PartialBatchKey> {
```

Comment:
this is pretty duplicative of finish, which is a bit of a bummer. I think `finish` could be implemented in terms of this, but my instinct is that it'd be nice to only have the one that accepts owership of self, so we don't have to worry about reuse

I haven't yet gotten to where this is used, so could be a totally naive comment

Reply:
oh I think I see a way to restructure this to call `finish` only at the end. I'll need to update how we calculate the run offsets (or `Vec<Vec<PartialBatchKey>>`), but that shouldn't be bad

Reply:
removed `flush`!